api/spec/requests/spree/api/customer_returns_controller_spec.rb

```diff
@@ -0,0 +1,113 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+module Spree
+  describe Api::CustomerReturnsController, type: :request do
+    let!(:order) { create(:shipped_order) }
+    let(:attributes) { [:id, :number, :stock_location_id] }
+
+    before do
+      stub_authentication!
+    end
+
+    context "as a non admin" do
+      before do
+        allow_any_instance_of(Order).to receive_messages user: create(:user)
+      end
+
+      it "cannot see any customer returns" do
+        get spree.api_order_customer_returns_path(order)
+        assert_unauthorized!
+      end
+
+      it "cannot see a single customer return" do
+        get spree.api_order_customer_return_path(order, 1)
+        assert_unauthorized!
+      end
+
+      it "cannot learn how to create a new customer return" do
+        get spree.new_api_order_customer_return_path(order)
+        assert_unauthorized!
+      end
+
+      it "cannot update a customer return" do
+        put spree.api_order_customer_return_path(order, 0)
+        assert_unauthorized!
+      end
+
+      it "cannot create a new customer return" do
+        post spree.api_order_customer_returns_path(order)
+        assert_unauthorized!
+      end
+    end
+
+    context "as an admin" do
+      sign_in_as_admin!
+
+      it "can show customer return" do
+        customer_return = FactoryBot.create(:customer_return)
+        get spree.api_order_customer_return_path(customer_return.order, customer_return.id)
```

Comment:
Can you add empty lines between the setup, exercise and verify phases of this and all other tests? It makes them much easier to read. Thanks!

Reply:
Done - let me know how that looks!